crates/oxc_linter/src/rule.rs

```diff
@@ -15,7 +15,9 @@ pub trait Rule: Sized + Default + Debug {
         Self::default()
     }
 
-    fn run<'a>(&self, node: &AstNode<'a>, _ctx: &LintContext<'a>);
+    fn run_once(&self, _ctx: &LintContext<'_>) {}
+
```

Comment:
Maybe we should have a `run_on_symbol` function instead so we can iterate through the symbols once for all the `no-xx-assign` rules.

Reply:
API wise I prefer `run_once` to keep it simple, but it may be a bit slower to iterate the symbol table for a bunch of rules.